src/server/main_service.cc

```diff
@@ -2382,23 +2377,45 @@ void Service::Pubsub(CmdArgList args, const CommandContext& cmd_cntx) {
         "NUMSUB [<channel> <channel...>]",
         "\tReturns the number of subscribers for the specified channels, excluding",
         "\tpattern subscriptions.",
+        "SHARDCHANNELS [pattern]",
+        "\tReturns a list of active shard channels, optionally matching the specified pattern ",
+        "(default: '*').",
+        "SHARDNUMSUB [<channel> <channel...>]",
+        "\tReturns the number of subscribers for the specified shard channels, excluding",
+        "\tpattern subscriptions.",
         "HELP",
         "\tPrints this help."};
 
     rb->SendSimpleStrArr(help_arr);
     return;
   }
 
-  if (subcmd == "CHANNELS") {
+  auto cmd_err = [&]() {
+    auto err = absl::StrCat("PUBSUB ", subcmd, " is not supported",
+                            IsClusterEnabled() ? " in cluster mode yet" : " in non cluster mode");
```

Comment:
@kostasrim I think that the check should be IsClusterEnabledOrEmulated() right?
cause we should enable cluster commands in emulated mode

Reply:
@mkaruza Dragonfly has emulated cluster mode which runs a single shard and in this state cluster and non cluster client are allowed.
This means that we should allow global pubsub command to run both on non cluster and emulated mode and sharded pubsub commands to run on both cluster and emulated mode, hence the check to send the error below should be fixed